ZeldaFullEditor/Gui/ExtraForms/ZsprImporter.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                byte version = br.ReadByte();
+                int checksum = br.ReadInt32();
+                int dataStart = br.ReadInt32();
+                ushort length = br.ReadUInt16();
+                int dataPalStart = br.ReadInt32();
+                ushort lengthPal = br.ReadUInt16();
+                ushort sprType = br.ReadUInt16();
+                br.ReadByte(); br.ReadByte(); br.ReadByte(); br.ReadByte(); br.ReadByte(); br.ReadByte(); // 6 empty bytes
+
+                long npos = br.BaseStream.Position;
+
+                byte b1 = br.ReadByte();
+                byte b2 = br.ReadByte();
+                int count = 2;
+                while (true)
+                {
+                    b1 = br.ReadByte();
+                    b2 = br.ReadByte();
+                    if (b1 == 0 && b2 == 0)
+                    {
+                        break;
+                    }
+
+                    count += 2;
+                }
+                
+                br.BaseStream.Seek(npos, SeekOrigin.Begin);
+                byte[] displayTextBytes = br.ReadBytes(count);
+                string displayText = Encoding.Unicode.GetString(displayTextBytes);
+                br.ReadByte(); // burn the 2 zero bytes
+                br.ReadByte();
+
+                npos = br.BaseStream.Position;
+
+                b1 = br.ReadByte();
+                b2 = br.ReadByte();
+                count = 2;
+                while (true)
+                {
+                    b1 = br.ReadByte();
+                    b2 = br.ReadByte();
+                    if (b1 == 0 && b2 == 0)
+                    {
+                        break;
+                    }
+
+                    count += 2;
+                }
+
+                br.BaseStream.Seek(npos, SeekOrigin.Begin);
+                byte[] displayAuthorBytes = br.ReadBytes(count);
+                string author = Encoding.Unicode.GetString(displayAuthorBytes);
+                //string displayText = br.ReadString();
+                //string author = br.ReadString();
+
+                labelInfos.Text = displayText + "\r\n" + author + "\r\n" + version.ToString();
+
+
+
+                br.BaseStream.Seek(dataStart, SeekOrigin.Begin);
+                gfxData = br.ReadBytes(0x7000);
+                bpp8Data = GFX.SnesTilesToPc8bppTiles(gfxData, 0x380, 4);
+
+                ptrImage.Draw8bppTiles(0, 0, bpp8Data, 16, 0, 0);
+
+
+                br.BaseStream.Seek(dataPalStart, SeekOrigin.Begin);
+                pal1[0] = Color.FromArgb(220, 220, 220);
+                for (int i = 0; i < 15; i++)
+                {
+                    pal1[i+1] = GFX.getColor(br.ReadInt16());
+                    ptrImage.UpdatePalettes(pal1);
+
+                }
+                for (int i = 0; i < 15; i++)
+                {
+                    pal1[i + 17] = GFX.getColor(br.ReadInt16());
```

Comment:
These offsets get us the current armor palettes i'm assuming? maybe a constant would be good here too